docs/source/depth_estimation.mdx

```diff
@@ -0,0 +1,214 @@
+```
+
+<div class="flex justify-center">
+    <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/datasets/depth_est_sample.png">
+</div>
+
+Now take a look at its corresponding depth map:
+
+```py
+>>> train_dataset[index]["depth_map"]
+```
+
+<div class="flex justify-center">
+    <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/datasets/depth_est_target.png">
+</div>
+
+It's all black! You'll need to add some color to the depth map to visualize it properly (the utility below is taken from the [FastDepth repository](https://github.com/dwofk/fast-depth/blob/master/utils.py)).
+
+```py 
+>>> import numpy as np
+>>> import matplotlib.pyplot as plt
+
+>>> cmap = plt.cm.viridis
+
+>>> def colored_depthmap(depth, d_min=None, d_max=None):
+...     if d_min is None:
+...         d_min = np.min(depth)
+...     if d_max is None:
+...         d_max = np.max(depth)
+...     depth_relative = (depth - d_min) / (d_max - d_min)
+...     return 255 * cmap(depth_relative)[:,:,:3]
+
+>>> def show_depthmap(depth_map):
+...    if not isinstance(depth_map, np.ndarray):
+...        depth_map = np.array(depth_map)
+...    if depth_map.ndim == 3:
+...        depth_map = depth_map.squeeze()
+
+...    d_min = np.min(depth_map)
+...    d_max = np.max(depth_map)
+...    depth_map = colored_depthmap(depth_map, d_min, d_max)
+
+...    plt.imshow(depth_map.astype("uint8"))
+...    plt.axis("off")
+...    plt.show()
+
+>>> show_depthmap(train_dataset[index]["depth_map"])
+```
+
+<div class="flex justify-center">
+    <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/datasets/depth_est_target_viz.png">
+</div>
+
+You can also visualize a collage of different images and the corresponding depth maps.
+
+```py
+>>> def merge_into_row(input_image, depth_target):
+...     if not isinstance(input_image, np.ndarray):
+...         input_image = np.array(input_image)
+...
+...     d_min = np.min(depth_target)
+...     d_max = np.max(depth_target)
+...     depth_target_col = colored_depthmap(depth_target, d_min, d_max)
+...     img_merge = np.hstack([input_image, depth_target_col])
+...
+...     return img_merge
+
+>>> random_indices = np.random.choice(len(train_dataset), 9).tolist()
+>>> plt.figure(figsize=(15, 6))
+>>> for i, idx in enumerate(random_indices):
+...     ax = plt.subplot(3, 3, i + 1)
+...     image_viz = merge_into_row(
+...         train_dataset[idx]["image"], train_dataset[idx]["depth_map"]
+...     )
+...     plt.imshow(image_viz.astype("uint8"))
+...     plt.axis("off")
+```
+
+<div class="flex justify-center">
+    <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/datasets/depth_est_collage.png">
+</div>
+
+Now apply some augmentations with `albumentations`. The augmentation transformations include:
+
+* Random cropping 
+* Random brightness and contrast 
+* Random gamma correction 
+* Random hue saturation
```

Comment:
I had to drop horizontal flipping. Here's why. 

The behavior seems to be unstable with `datasets` when applied via `set_transform()`. Consider the following as an example. 

<img width="1533" alt="debug_augmentations" src="https://user-images.githubusercontent.com/22957388/209280992-08154743-9855-4eb2-842a-dab25432072a.png">

You'll notice that subplots tagged as "wrong" have flipped images but their corresponding depth maps are not flipped. This is not a consistent case, though. Subplots tagged as "right", for example. You'll notice that the images and their depth maps have both been flipped there (compare them to the originals and you should be able to spot the difference). 

I have verified if there's a problem with `albumentations` but that doesn't seem to be the case. Refer to this notebook: https://gist.github.com/sayakpaul/63ae5f7971f634894ab37e76e6b3879c. You'll notice that I took individual samples from the dataset and passed them through the `aug` chain and the results were fine. 

Have you experienced this before? Is anything obviously wrong you can spot in the code? 

Let me know if anything's unclear. 

@nateraw @lhoestq

Reply:
I'm unable to reproduce this issue 🙁 . Take a look at this [simplified notebook](https://gist.github.com/nateraw/434fd94de897d76f449a805fcc3ec623).

Reply:
Yes @nateraw. But it doesn't seem to be consistent, as I stated here:

> You'll notice that subplots tagged as "wrong" have flipped images but their corresponding depth maps are not flipped. This is not a consistent case, though. Subplots tagged as "right", for example.

Sometimes, the augmented depth maps are correct, i.e., the flip transformation has been correctly applied, and sometimes, that doesn't seem to be the case. You'll notice in the collage [here](https://github.com/huggingface/datasets/pull/5379#discussion_r1056065300).

Reply:
Yea I noticed that, but I ran it over and over and never got that behavior. Do you have a minimal reproduction of the behavior?

Reply:
I don't. Strangely, it only comes up when I try to put it inside a collection of images.

I will try to reproduce the behavior minimally and let you know here.

Reply:
My god! Thanks for saving me!